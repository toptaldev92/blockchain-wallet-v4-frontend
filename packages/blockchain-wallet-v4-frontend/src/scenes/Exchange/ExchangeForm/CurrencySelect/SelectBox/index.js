import React from 'react'
import styled from 'styled-components'
import { pathOr, prop, toLower } from 'ramda'

import { Icon } from 'blockchain-info-components'
import { SelectBox } from 'components/Form'

const ExchangeSelect = styled(SelectBox)`
  .bc__control {
    border: none;
    :hover {
      border: none;
    }
    :active {
      border: none;
    }
  }

  .bc__value-container {
    padding: 0;
  }
  .bc__option {
    padding: 8px 0;
    font-weight: 600;
  }

  .bc__single-value {
    color: ${props => props.theme.white};
  }
`

const DisplayWrapper = styled.div`
  border-radius: 3px;
  background-color: ${props => props.theme[props.coin]};
  min-height: 30px;
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
  height: 100%;
  padding: 5px;
  box-sizing: border-box;
  cursor: pointer;
`
const ItemWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
  height: 100%;
  padding: 5px;
  box-sizing: border-box;
  cursor: pointer;
  min-width: 0;
`
const Text = styled.span`
  position: relative;
  font-family: 'Inter', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto,
    Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-weight: 500;
  font-size: 12px;
  text-transform: ${props => (props.uppercase ? 'uppercase' : 'none')};
  font-style: normal;
  cursor: pointer;
  width: 100%;
  padding-left: 4px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
`

const DisplayIcon = styled(Icon)`
  font-size: 26px;
  color: ${props => props.theme.white};
`

const selectItemIconColor = props => props.theme[props.coin]

const ItemIcon = styled(Icon)`
  font-size: 20px;
  color: ${selectItemIconColor} !important;
`

const renderDisplay = (props, children) => {
  const coin = pathOr('', ['value', 'coin'], props)
  const icon = pathOr('', ['value', 'icon'], props)
  return (
    <DisplayWrapper className={props.className} coin={toLower(coin)}>
      {<DisplayIcon name={icon} />}
      <Text>{children}</Text>
    </DisplayWrapper>
  )
}

const renderItem = item => {
  const coin = pathOr('', ['value', 'coin'], item)
  const icon = pathOr('', ['value', 'icon'], item)
  const isSelected = prop('isSelected', item)
  return (
    <ItemWrapper>
      {<ItemIcon coin={toLower(coin)} isSelected={isSelected} name={icon} />}
      <Text>{item.text}</Text>
    </ItemWrapper>
  )
}

const SelectBoxExchange = props => (
  <ExchangeSelect
    {...props}
    searchEnabled={false}
    hideIndicator={true}
    templateDisplay={renderDisplay}
    templateItem={renderItem}
  />
)

export default SelectBoxExchange
