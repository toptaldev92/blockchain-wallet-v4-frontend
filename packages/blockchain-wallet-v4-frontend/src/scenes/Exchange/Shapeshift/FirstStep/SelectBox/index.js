import React from 'react'

import { SelectBox } from 'components/Form'

class SelectBoxAccounts extends React.Component {
  render () {
    return <SelectBox />
  }
}

export default SelectBoxAccounts
