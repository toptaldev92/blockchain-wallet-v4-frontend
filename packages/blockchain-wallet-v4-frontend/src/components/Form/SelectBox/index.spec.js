import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import SelectBox from './index.js'
jest.mock('blockchain-info-components', () => ({ SelectInput: 'select-input' }))

describe('SelectBox', () => {
  it('renders correctly', () => {
    const props = { meta: {} }
    const component = renderer.create(<SelectBox {...props} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
