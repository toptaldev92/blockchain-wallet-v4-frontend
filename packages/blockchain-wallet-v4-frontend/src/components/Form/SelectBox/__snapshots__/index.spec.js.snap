// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectBox renders correctly 1`] = `
<styled.div
  data-e2e="dropdownSelect"
>
  <SelectInput
    elements={
      Array [
        Object {
          "group": "fakeG",
          "items": Array [
            Object {
              "text": "1",
              "value": 1,
            },
            Object {
              "text": "2",
              "value": 2,
            },
            Object {
              "text": "3",
              "value": 3,
            },
          ],
        },
      ]
    }
    errorState="initial"
    getRef={[Function]}
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    onFocus={[MockFunction]}
    onKeyDown={[Function]}
    value=""
  />
</styled.div>
`;
